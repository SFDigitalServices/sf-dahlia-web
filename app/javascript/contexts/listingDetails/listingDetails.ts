interface ListingDetails {
  /**  */
  uid: string

  /**  */
  email: string

  /**  */
  created_at: Date

  /**  */
  updated_at: Date
}

interface ListingDetailsData {
  data: any
  success: boolean
}

export { ListingDetails, ListingDetailsData }
