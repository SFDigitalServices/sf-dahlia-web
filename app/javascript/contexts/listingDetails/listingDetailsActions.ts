import { createAction } from "typesafe-actions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"

export enum ListingDetailsActions {
  StartFetchingUnits = "SetFetchingUnits",
  FinishFetchingUnits = "FinishFetchingUnits",
  SetUnits = "SetUnits",
  SetAmiCharts = "SetAmiCharts",
  StartFetchingAmiCharts = "SetFetchingAmiCharts",
  SetFetchedAmiCharts = "SetFetchedAmiCharts",
  SetFetchingAmiChartsError = "SetFetchingAmiChartsError",
  SetFetchingUnitsError = "SetFetchingUnitsError",
  FinishFetchingAmiCharts = "FinishFetchingAmiCharts",
}

export const startFetchingUnits = createAction(ListingDetailsActions.StartFetchingUnits)()
export const setUnits = createAction(ListingDetailsActions.SetUnits)()
export const setAmiCharts = createAction(ListingDetailsActions.SetAmiCharts)()
export const finishFetchingAmiCharts = createAction(ListingDetailsActions.FinishFetchingAmiCharts)<
  RailsAmiChart[]
>()
export const setFetchedAmiCharts = createAction(ListingDetailsActions.SetFetchedAmiCharts)()
export const startFetchingAmiCharts = createAction(ListingDetailsActions.StartFetchingAmiCharts)()
export const finishFetchingUnits = createAction(ListingDetailsActions.FinishFetchingUnits)<
  RailsUnit[]
>()
export const setFetchingUnitsError = createAction(
  ListingDetailsActions.SetFetchingUnitsError
)<Error>()
export const setFetchingAmiChartsError = createAction(
  ListingDetailsActions.SetFetchingAmiChartsError
)<Error>()
