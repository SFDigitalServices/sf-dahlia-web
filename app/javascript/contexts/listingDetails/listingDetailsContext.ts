import { createContext } from "react"
import RailsUnit from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"

export type ContextProps = {
  units: RailsUnit[]
  amiCharts: RailsAmiChart[]
  fetchedAmiCharts: boolean
  fetchingAmiCharts: boolean
  fetchedUnits: boolean
  fetchingUnits: boolean
  fetchUnits: (listingId: string) => void
  fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => void
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

export default createContext<Partial<ContextProps>>({})
