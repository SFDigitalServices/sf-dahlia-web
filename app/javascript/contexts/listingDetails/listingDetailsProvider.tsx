import React, { useReducer } from "react"
import ListingDetailsReducer from "./listingDetailsReducer"
import ListingDetailsContext, { ContextProps } from "./listingDetailsContext"
import { getUnits, getAmiCharts } from "../../api/listingApiService"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"
import {
  finishFetchingUnits,
  finishFetchingAmiCharts,
  startFetchingAmiCharts,
  startFetchingUnits,
  setFetchingAmiChartsError,
  setFetchingUnitsError,
} from "./listingDetailsActions"

interface ListingDetailsProviderProps {
  children?: React.ReactNode
}

const ListingDetailsProvider = (props: ListingDetailsProviderProps) => {
  const [state, dispatch] = useReducer(ListingDetailsReducer, {
    units: [],
    amiCharts: [],
    fetchingUnits: false,
    fetchedUnits: false,
    fetchingAmiCharts: false,
    fetchedAmiCharts: false,
    fetchingAmiChartsError: null,
    fetchingUnitsError: null,
  })

  const contextValues: ContextProps = {
    units: state.units,
    amiCharts: state.amiCharts,
    fetchedAmiCharts: state.fetchedAmiCharts,
    fetchingAmiCharts: state.fetchingAmiCharts,
    fetchedUnits: state.fetchedUnits,
    fetchingUnits: state.fetchingUnits,
    fetchingAmiChartsError: state.fetchingAmiChartsError,
    fetchingUnitsError: state.fetchingUnitsError,
    fetchUnits: (listingId: string) => {
      dispatch(startFetchingUnits())
      getUnits(listingId)
        .then((units: RailsUnit[]) => {
          dispatch(finishFetchingUnits(units))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingUnitsError(error))
        })
    },
    fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => {
      dispatch(startFetchingAmiCharts())
      getAmiCharts(chartsToFetch)
        .then((amiCharts: RailsAmiChart[]) => {
          dispatch(finishFetchingAmiCharts(amiCharts))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingAmiChartsError(error))
        })
    },
  }

  return (
    <ListingDetailsContext.Provider value={contextValues}>
      {props.children}
    </ListingDetailsContext.Provider>
  )
}

export default ListingDetailsProvider
