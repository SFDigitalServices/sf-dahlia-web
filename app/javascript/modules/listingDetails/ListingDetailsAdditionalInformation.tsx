import React from "react"
import { LinkButton, ListingDetailItem, t } from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { TextTruncate } from "../../components/TextTruncate"
import { isHabitatListing, isSale } from "../../util/listingUtil"
import { stripMostTags } from "../../util/filterUtil"
import { getTranslatedString } from "../../util/languageUtil"

export interface ListingDetailsAdditionalInformationProps {
  listing: RailsListing
  imageSrc: string
}

export const ListingDetailsAdditionalInformation = ({
  listing,
  imageSrc,
}: ListingDetailsAdditionalInformationProps) => {
  const getCommissionString = () => {
    return listing.Realtor_Commission_Unit === "percent"
      ? t("listings.realtorCommissionPercentage", {
          percentage: listing.Realtor_Commission_Amount,
        })
      : `$${listing.Realtor_Commission_Amount.toLocaleString()}`
  }

  return (
    <ListingDetailItem
      imageAlt={""}
      imageSrc={imageSrc}
      title={t("listings.additionalInformation.header")}
      subtitle={t("listings.additionalInformation.subheader")}
    >
      <div className="listing-detail-panel">
        {listing.Listing_Other_Notes && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.specialNotes")}</h3>
            <TextTruncate
              className="primary-lighter-markup-link translate"
              buttonClassName="text-blue-700"
              text={stripMostTags(
                getTranslatedString(
                  listing.Listing_Other_Notes,
                  "Listing_Other_Notes",
                  listing.translations
                )
              )}
            />
          </div>
        )}
        {(!!listing.Required_Documents || isSale(listing)) && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.requiredDocuments")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-700"
                text={stripMostTags(
                  getTranslatedString(
                    listing.Required_Documents,
                    "Required_Documents",
                    listing.translations
                  )
                )}
              />
            </div>
            {isSale(listing) && !isHabitatListing(listing) && (
              <div className="text-xs mt-4">
                <TextTruncate
                  className="primary-lighter-markup-link translate"
                  buttonClassName="text-blue-700"
                  text={t("listings.requiredDocumentsAfterApplying", {
                    url: "https://sfmohcd.org/after-homebuyer-lottery",
                  })}
                />
              </div>
            )}
          </div>
        )}
        {listing.Legal_Disclaimers && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.importantProgramRules")}</h3>
            <div className="text-xs">
              <TextTruncate
                text={stripMostTags(
                  getTranslatedString(
                    listing.Legal_Disclaimers,
                    "Legal_Disclaimers",
                    listing.translations
                  )
                )}
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
              />
            </div>
          </div>
        )}
        {listing.CC_and_R_URL && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.cc&r")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
                text={t("listings.cc&rDescription")}
              />
              <br />
              <LinkButton href={listing.CC_and_R_URL} className={"mt-4"} newTab={true}>
                {t("listings.downloadPdf")}
              </LinkButton>
            </div>
          </div>
        )}
        {isSale(listing) && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.realtorCommission")}</h3>
            <div className="text-xs">
              {listing.Allows_Realtor_Commission ? (
                <>
                  <span className={"font-bold"}>{t("listings.realtorCommissionHeader")}</span>
                  {getCommissionString()}
                </>
              ) : (
                t("listings.realtorCommissionNotEligible")
              )}
              {listing.Realtor_Commission_Info && (
                <div className={"mt-4"}>
                  <span className={"font-bold"}>{t("listings.realtorCommissionHowTo")}</span>
                  <span>
                    <TextTruncate
                      className="primary-lighter-markup-link translate"
                      buttonClassName="text-blue-500"
                      text={stripMostTags(
                        getTranslatedString(
                          listing.Realtor_Commission_Info,
                          "Realtor_Commission_Info",
                          listing.translations
                        )
                      )}
                    />
                  </span>
                </div>
              )}
            </div>
          </div>
        )}
        {/* TODO: implement once we've established needs for sales listings
              {listing.isSale && (
                <div className="info-card bg-gray-100 border-0">
                  <h3 className="text-serif-xl">For the Buyer's Realtor</h3>
                  {listing.Allows_Realtor_Commission ? (
                    display realtor_commission_header
                    realtorComissionMessage
                    {listing.Realtor_Commission_Info && realtor_commission_how_to}
                  ) : display realtor_commission_not_eligible message}
                </div>
              )}
            */}
        {listing.Repricing_Mechanism && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.rePricing")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
                text={stripMostTags(
                  getTranslatedString(
                    listing.Repricing_Mechanism,
                    "Repricing_Mechanism",
                    listing.translations
                  )
                )}
              />
            </div>
          </div>
        )}
      </div>
    </ListingDetailItem>
  )
}
