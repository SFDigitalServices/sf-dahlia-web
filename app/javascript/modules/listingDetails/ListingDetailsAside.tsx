import React from "react"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsInfoSession } from "../listingDetailsAside/ListingDetailsInfoSession"
import { ListingDetailsApply } from "../listingDetailsAside/ListingDetailsApply"
import { ListingDetailsProcess } from "../listingDetailsAside/ListingDetailsProcess"
import { isOpen } from "../../util/listingUtil"
import { ListingDetailsApplicationDate } from "../listingDetailsAside/ListingDetailsApplicationDate"

export interface ListingDetailsSidebarProps {
  listing: RailsListing
}

export const ListingDetailsAside = ({ listing }: ListingDetailsSidebarProps) => {
  const isApplicationOpen = isOpen(listing)

  return (
    <aside className="w-full static md:absolute md:right-0 md:w-1/3 md:top-0 sm:w-2/3 md:ml-2 h-full md:border border-solid bg-white">
      <div className="hidden md:block">
        <ListingDetailsApplicationDate listing={listing} />
        {isApplicationOpen && <ListingDetailsInfoSession listing={listing} />}
        <ListingDetailsApply listing={listing} />
        <ListingDetailsProcess listing={listing} />
      </div>
    </aside>
  )
}
