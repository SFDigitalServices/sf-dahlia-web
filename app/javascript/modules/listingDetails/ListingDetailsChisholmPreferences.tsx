import React from "react"
import { ListSection, t } from "@bloom-housing/ui-components"
import { renderInlineMarkup, getSfGovUrl } from "../../util/languageUtil"

export const ListingDetailsChisholmPreferences = ({
  isEducatorOne,
}: {
  isEducatorOne: boolean
}) => (
  <ListSection
    title={t("listings.lottery.title")}
    subtitle={
      <div className="space-y-4">
        <p>{t("listings.customListingType.educator.preferences.part1")}</p>
        <p>
          {renderInlineMarkup(t("listings.customListingType.educator.preferences.part2"), "<b>")}
          <br />
          {isEducatorOne
            ? t("listings.customListingType.educator.preferences.part3")
            : t("listings.customListingType.educator.preferences.part3.scv2")}
        </p>
        <ul className="list-disc ml-7">
          <li>
            {renderInlineMarkup(
              t("listings.customListingType.educator.preferences.part4a", {
                chisholmLotteryLink: getSfGovUrl(
                  "https://sf.gov/information/learn-how-lottery-works-shirley-chisholm-village",
                  10493
                ),
              })
            )}
          </li>
          <li>
            {renderInlineMarkup(
              t("listings.customListingType.educator.preferences.part4b", {
                preferencesLink: getSfGovUrl(
                  "https://sf.gov/information/learn-about-housing-lottery-preference-programs",
                  3274
                ),
              })
            )}
          </li>
        </ul>
      </div>
    }
  />
)
