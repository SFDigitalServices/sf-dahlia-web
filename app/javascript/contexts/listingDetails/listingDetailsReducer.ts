import { createReducer } from "typesafe-actions"
import { ListingDetailsActions } from "./listingDetailsActions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"
import { AmiChartData } from "./listingDetailsContext"

type ListingDetailsState = {
  fetchingUnits: boolean
  fetchedUnits: boolean
  fetchingAmiCharts: boolean
  fetchedAmiCharts: boolean
  units: RailsUnit[]
  amiChartData: AmiChartData
  amiCharts: RailsAmiChart[]
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState, {
  [ListingDetailsActions.StartFetchingUnits]: (state) => {
    return { ...state, fetchingUnits: true }
  },
  [ListingDetailsActions.StartFetchingAmiCharts]: (state) => {
    return { ...state, fetchingAmiCharts: true }
  },
  [ListingDetailsActions.FinishFetchingUnits]: (state, { payload: units }) => {
    const amiChartYears: Array<number> = []
    const amiChartTypes = []
    const amiChartPercentages = []

    units.forEach((unit: RailsUnit) => {
      const amiChartYear = amiChartYears.find((year) => {
        return year === unit.AMI_chart_year
      })

      const amiChartType = amiChartTypes.find((type) => {
        return type === unit.AMI_chart_type
      })

      const amiChartPercentage = amiChartPercentages.find((type) => {
        return type === unit.Max_AMI_for_Qualifying_Unit
      })

      if (!amiChartYear) {
        amiChartYears.push(unit.AMI_chart_year)
      }

      if (!amiChartType) {
        amiChartTypes.push(unit.AMI_chart_type)
      }

      if (!amiChartPercentage) {
        amiChartPercentages.push(unit.Max_AMI_for_Qualifying_Unit)
      }
    })

    return {
      ...state,
      fetchingUnits: false,
      fetchedUnits: true,
      units,
      amiChartData: {
        years: amiChartYears,
        percentages: amiChartPercentages,
        types: amiChartTypes,
      },
    }
  },
  [ListingDetailsActions.FinishFetchingAmiCharts]: (state, { payload: amiCharts }) => {
    return {
      ...state,
      fetchingAmiCharts: false,
      fetchedAmiCharts: true,
      amiCharts: amiCharts?.map((amiChart: RailsAmiChart) => {
        return {
          ...amiChart,
          chartType: amiChart?.values[0]?.chartType,
        }
      }),
    }
  },
})

export default ListingDetailsReducer
