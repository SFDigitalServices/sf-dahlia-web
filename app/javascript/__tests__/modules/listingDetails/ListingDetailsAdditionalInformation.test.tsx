import React from "react"
import { render } from "@testing-library/react"
import { ListingDetailsAdditionalInformation } from "../../../modules/listingDetails/ListingDetailsAdditionalInformation"
import { closedRentalListing } from "../../data/RailsRentalListing/listing-rental-closed"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"

describe("ListingDetailsAdditionalInformation", () => {
  it("displays additional information section for a sale listing", () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })

    const { asFragment } = render(
      <ListingDetailsAdditionalInformation listing={openSaleListing} imageSrc={"/image-url"} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays additional information section for a rental listing", () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })

    const { asFragment } = render(
      <ListingDetailsAdditionalInformation listing={closedRentalListing} imageSrc={"/image-url"} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays realtor commission section if Allows_Realtor_Commission is true for a sales listing", () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })

    const listing = {
      ...openSaleListing,
      Allows_Realtor_Commission: true,
    }

    const { asFragment } = render(
      <ListingDetailsAdditionalInformation listing={listing} imageSrc={"/image-url"} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays realtor commission text if Realtor_Commission_Amount and Realtor_Commission_Unit are present for a sales listing", () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })

    const listing = {
      ...openSaleListing,
      Allows_Realtor_Commission: true,
      Realtor_Commission_Amount: 3,
      Realtor_Commission_Unit: "percent",
    }

    const { asFragment } = render(
      <ListingDetailsAdditionalInformation listing={listing} imageSrc={"/image-url"} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
