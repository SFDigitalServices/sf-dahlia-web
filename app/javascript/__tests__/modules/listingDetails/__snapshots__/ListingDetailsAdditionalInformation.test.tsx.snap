// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsAdditionalInformation displays additional information section for a rental listing 1`] = `
<DocumentFragment>
  <li
    class="responsive-content-item "
  >
    <header
      class="detail-header "
    >
      <span
        class="detail-header__image-container"
      >
        <img
          alt=""
          class="detail-header__image "
          src="/image-url"
        />
      </span>
      <hgroup
        class="detail-header__hgroup"
      >
        <h2
          class="detail-header__title"
        >
          Additional Information
        </h2>
        <span
          class="detail-header__subtitle"
        >
          Required documents and selection criteria
        </span>
        <span
          class="ui-icon ui-medium"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </hgroup>
    </header>
    <div
      class="listing-detail-panel"
    >
      <div
        class="info-card bg-gray-100 border-0"
      >
        <h3
          class="text-serif-xl"
        >
          Required Documents
        </h3>
        <div
          class="text-xs"
        >
          <div
            class="text-xs inline primary-lighter-markup-link translate"
          >
            <span>
              Lottery winners will be required to fill out a building application and provide a copy of your current credit report, 3 most recent paystubs, current tax returns and W-2, and 3 most recent bank statements.
            </span>
          </div>
        </div>
      </div>
      <div
        class="info-card bg-gray-100 border-0"
      >
        <h3
          class="text-serif-xl"
        >
          Important Program Rules
        </h3>
        <div
          class="text-xs"
        >
          <div
            class="text-xs inline primary-lighter-markup-link translate"
          >
            <span>
              All BMR renters must review and acknowledge the 
              <a
                href="http://sf-moh.org/index.aspx?page=295"
                target="_blank"
              >
                Inclusionary Affordable Housing Program Monitoring and Procedures Manual 2013
              </a>
               that governs this property upon the signing of a lease for a BMR unit. 
              <br />
               
              <br />
              Applicants should be informed that BMR rental units in some buildings may convert to ownership units in the future.  In the case of conversion, BMR renters will be afforded certain rights as explained in the 
              <a
                href="http://sf-moh.org/index.aspx?page=295"
                target="_blank"
              >
                Inclusionary Affordable Housing Program Monitoring and Procedures Manual 2013
              </a>
              . Applicants should inquire with the building contact person listed above to determine if the building has a minimum number of years that it must remain a rental building.  (Some buildings may have such restrictions based on government sources of financing for their building.) Most buildings may have no restrictions on conversion at all. 
              <br />
               
              <br />
              It is also important to note that units governed by the Inclusionary Housing Program are NOT governed by the San Francisco Rent Ordinance (also known as “rent control”). Among other rules, rents may increase beyond increases allowed under “rent control.”  Please see the 
              <a
                href="http://sf-moh.org/index.aspx?page=295"
                target="_blank"
              >
                Inclusionary Affordable Housing Program Monitoring and Procedures Manual 2013
              </a>
               for more information.
            </span>
          </div>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`ListingDetailsAdditionalInformation displays additional information section for a sale listing 1`] = `
<DocumentFragment>
  <li
    class="responsive-content-item "
  >
    <header
      class="detail-header "
    >
      <span
        class="detail-header__image-container"
      >
        <img
          alt=""
          class="detail-header__image "
          src="/image-url"
        />
      </span>
      <hgroup
        class="detail-header__hgroup"
      >
        <h2
          class="detail-header__title"
        >
          Additional Information
        </h2>
        <span
          class="detail-header__subtitle"
        >
          Required documents and selection criteria
        </span>
        <span
          class="ui-icon ui-medium"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </hgroup>
    </header>
    <div
      class="listing-detail-panel"
    >
      <div
        class="info-card bg-gray-100 border-0"
      >
        <h3
          class="text-serif-xl"
        >
          Special Notes
        </h3>
        <div
          class="text-xs inline primary-lighter-markup-link translate"
        >
          <span>
            Some other excessively formatted listing notes.
          </span>
        </div>
      </div>
      <div
        class="info-card bg-gray-100 border-0"
      >
        <h3
          class="text-serif-xl"
        >
          Required Documents
        </h3>
        <div
          class="text-xs"
        >
          <div
            class="text-xs inline primary-lighter-markup-link translate"
          >
            <span>
              Proof of residence

              <br />
              Bank statements

              <br />
              Concert tickets
            </span>
          </div>
        </div>
      </div>
      <div
        class="info-card bg-gray-100 border-0"
      >
        <h3
          class="text-serif-xl"
        >
          Important Program Rules
        </h3>
        <div
          class="text-xs"
        >
          <div
            class="text-xs inline primary-lighter-markup-link translate"
          >
            <span>
              Some important program rules with bullets:
              <br />
              Rule oneRule two
            </span>
          </div>
        </div>
      </div>
      <div
        class="info-card bg-gray-100 border-0"
      >
        <h3
          class="text-serif-xl"
        >
          Covenants, Conditions and Restrictions (CC&R's)
        </h3>
        <div
          class="text-xs"
        >
          <div
            class="text-xs inline primary-lighter-markup-link translate"
          >
            <span>
              The CC&R's explain the rules of the homeowners' association, and restrict how you can modify the property.
            </span>
          </div>
          <br />
          <a
            class="button mt-4"
            href="http://www.duckduckgo.com"
            target="_blank"
          >
            Download PDF
          </a>
        </div>
      </div>
      <div
        class="info-card bg-gray-100 border-0"
      >
        <h3
          class="text-serif-xl"
        >
          Resale Price Restrictions
        </h3>
        <div
          class="text-xs"
        >
          <div
            class="text-xs inline primary-lighter-markup-link translate"
          >
            <span>
              A BMR unit will be resold at a restricted price to a household that meets the first‐time homebuyer and income qualifications for the program. Please review the 
              <a
                href="http://sf-moh.org/index.aspx?page=295"
                target="_blank"
              >
                Inclusionary Housing Procedures Manual
              </a>
               for specific information.
            </span>
          </div>
        </div>
      </div>
      <div
        class="info-card bg-gray-100 border-0"
      >
        <h3
          class="text-serif-xl"
        >
          Housing Program
        </h3>
        <a
          class="text-xs"
          href="https://sfmohcd.org/for-buyers"
          target="_blank"
        >
          Below Market Rate
        </a>
      </div>
      <div
        class="info-card bg-gray-100 border-0"
      >
        <p
          class="text-xs"
        >
          Listing Updated: April 11, 2022
        </p>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
